lib/events/badge.js

```diff
@@ -47,42 +52,50 @@ function snowplowData(that, eventProperties, additionalProperties) {
   };
 }
 
-function badgeEvent(that, eventType, props) {
-  const eventProperties = { eventType };
-  const additionalProperties = {};
 
-  for (const prop in props) {
-    if (PROPERTIES.indexOf(prop) === -1) {
-      additionalProperties[prop] = props[prop];
-    } else {
-      eventProperties[prop] = props[prop];
-    }
+export default class Badge {
+  constructor(trkref) {
+    this.trkref = trkref;
   }
 
-  const data = snowplowData(that, eventProperties, additionalProperties);
-
-  validate(data, VALIDATIONS);
+  track(eventType, props) {
+    const properties = propertiesForEventType(eventType);
+    const eventProperties = { eventType };
+    const additionalProperties = {};
+
+    for (const prop in props) {
+      if (properties.indexOf(prop) === -1) {
+        additionalProperties[prop] = props[prop];
+      } else {
+        eventProperties[prop] = props[prop];
+      }
+    }
 
-  snowplow('trackUnstructEvent', {
-    schema: 'iglu:com.reevoo/badge_event/jsonschema/1-0-0',
-    data:   data,
-  });
-}
+    const data = snowplowData(this, eventProperties, additionalProperties);
```

Comment:
Does this work correctly? I know that sometimes (mainly working with React) you have to bind the `this` reference to the method in the constructor for example. I remember Jonny having some problems with this because it wasn't working as expected.

Example (in the constructor):

```javascript
this.track = this.track.bind(this);
```